app/src/test/java/org/simple/clinic/bp/entry/BloodPressureEntryLogicTest.kt

```diff
@@ -889,6 +890,133 @@ class BloodPressureEntrySheetLogicTest {
     verifyNoMoreInteractions(ui)
   }
 
+  @Test
+  fun `when a different user clicks on save while updating a BP, then the updated BP measurement should be saved with the user's ID and the corresponding facility ID`() {
+    // given
+    reset(userSession, facilityRepository)
+    val userFromDifferentFacility = PatientMocker.loggedInUser(uuid = UUID.fromString("4844b826-a162-49fe-b92c-962da172e86c"))
+    val differentFacility = PatientMocker.facility(uuid = UUID.fromString("f895b54f-ee32-4471-bc0c-a91b80368778"))
+
+    whenever(userSession.requireLoggedInUser()).doReturn(Observable.just(userFromDifferentFacility))
+    whenever(facilityRepository.currentFacility(userFromDifferentFacility)).doReturn(Observable.just(differentFacility))
+
+    val oldCreatedAt = LocalDate.of(1990, 1, 13).toUtcInstant(testUserClock)
+    val patientUuid = UUID.fromString("af92b081-0131-4f91-9c28-98da5737945b")
+    val existingBp = PatientMocker.bp(
+        uuid = patientUuid,
+        userUuid = user.uuid,
+        facilityUuid = facility.uuid,
+        systolic = 9000,
+        diastolic = 8999,
+        createdAt = oldCreatedAt,
+        updatedAt = oldCreatedAt,
+        recordedAt = oldCreatedAt
+    )
+
+    val newInputDate = LocalDate.of(1991, 2, 14)
+    whenever(bloodPressureRepository.measurement(existingBp.uuid)).doReturn(Observable.just(existingBp))
+    whenever(bloodPressureRepository.updateMeasurement(any())).doReturn(Completable.complete())
+    whenever(patientRepository.compareAndUpdateRecordedAt(any(), any())).doReturn(Completable.complete())
+
+    sheetCreatedForUpdate(existingBp.uuid)
+    uiEvents.run {
+      onNext(ScreenChanged(BP_ENTRY))
+      onNext(SystolicChanged("120"))
+      onNext(DiastolicChanged("110"))
+      onNext(SaveClicked)
+    }
+
+    uiEvents.run {
+      onNext(ScreenChanged(DATE_ENTRY))
+      onNext(DayChanged("14"))
+      onNext(MonthChanged("02"))
+      onNext(YearChanged("91"))
+      onNext(SaveClicked)
+    }
+
+    val newInputDateAsInstant = newInputDate.toUtcInstant(testUserClock)
+    val updatedBp = existingBp.copy(
+        systolic = 120,
+        diastolic = 110,
+        updatedAt = oldCreatedAt,
+        recordedAt = newInputDateAsInstant,
+        userUuid = userFromDifferentFacility.uuid,
+        facilityUuid = differentFacility.uuid
+    )
+    verify(bloodPressureRepository).updateMeasurement(updatedBp)
+    verify(patientRepository).compareAndUpdateRecordedAt(updatedBp.patientUuid, updatedBp.recordedAt)
+
+    verify(bloodPressureRepository, never()).saveMeasurement(any(), any(), any(), any(), any(), any())
+    verify(appointmentRepository, never()).markAppointmentsCreatedBeforeTodayAsVisited(any())
+    verify(ui).setBpSavedResultAndFinish()
+  }
+
+  @Test
+  fun `when done button is clicked by a user from a different facility in update BP entry, then save BP with entered date immediately`() {
```

Comment:
Isn't this testing the same thing as the earlier test? I can't tell the difference.

Reply:
One is clicking the **Save** button on the BP entry sheet. The other one is using the IME's *Done* button.

Reply:
But these events both get mapped to `SaveClicked` at the UI level, right?

Reply:
I also looked at the screen. There is no separate button on that sheet. The only way to save the BP is by the IME done button.

Reply:
I guess, then the test checks saving the BP from two different screens.

Reply:
Hmm, that's not the case as well. It's the `BP_ENTRY` screen on both. 😅

Reply:
Never mind, I just relooked at the tests and saw that the earlier test also changes the screen. Makes sense now.